source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -25,1139 +25,3434 @@
+ * Minimum: 1
+ *
+ * Maximum number of entries in the Neighbour Discovery cache of IPv6 addresses
+ * & MAC addresses
  */
-#ifndef ipconfigTCP_MAY_LOG_PORT
-    #define ipconfigTCP_MAY_LOG_PORT( xPort )    ( ( xPort ) != 23U )
-#endif
 
-/* Include all API's and code that is needed for the IPv4 protocol.
- * When defined as zero, the application should uses IPv6. */
-#ifndef ipconfigUSE_IPv4
-    #define ipconfigUSE_IPv4    ( 1 )
+#ifndef ipconfigND_CACHE_ENTRIES
+    #define ipconfigND_CACHE_ENTRIES    ( 24 )
 #endif
 
-/* Include all API's and code that is needed for the IPv6 protocol.
- * When defined as zero, the application should uses IPv4. */
-#ifndef ipconfigUSE_IPv6
-    #define ipconfigUSE_IPv6    ( 1 )
+#if ( ipconfigND_CACHE_ENTRIES < 1 )
+    #error ipconfigND_CACHE_ENTRIES must be at least 1
 #endif
 
-#if ( ipconfigUSE_IPv4 != 1 ) && ( ipconfigUSE_IPv6 != 1 )
-    #error "Invalid build configuration"
+#if ( ipconfigND_CACHE_ENTRIES > SIZE_MAX )
+    #error ipconfigND_CACHE_ENTRIES overflows a size_t
 #endif
 
+/*-----------------------------------------------------------------------*/
+
 /*
- * If defined this macro enables the APIs that are backward compatible
- * with single end point IPv4 version of the FreeRTOS+TCP library.
+ * ipconfigUSE_RA
+ *
+ * Type: BaseType_t ( ipconfigENABLE | ipconfigDISABLE )
+ *
+ * Include support for Router Advertisement (RA).
  */
-#ifndef ipconfigIPv4_BACKWARD_COMPATIBLE
-    #define ipconfigIPv4_BACKWARD_COMPATIBLE    0
-#endif
 
-/* Determine the number of clock ticks that the API's FreeRTOS_recv() and
- * FreeRTOS_recvfrom() must wait for incoming data. */
-#ifndef ipconfigSOCK_DEFAULT_RECEIVE_BLOCK_TIME
-    #define ipconfigSOCK_DEFAULT_RECEIVE_BLOCK_TIME    portMAX_DELAY
+#ifndef ipconfigUSE_RA
+    #define ipconfigUSE_RA    ipconfigENABLE
 #endif
 
-/* Determine the number of clock ticks that FreeRTOS_send() must wait
- * for space in the transmission buffer.
- * For FreeRTOS_sendto(), it limits how long the application
- * should wait for a network buffer to become available.
- */
-#ifndef ipconfigSOCK_DEFAULT_SEND_BLOCK_TIME
-    #define ipconfigSOCK_DEFAULT_SEND_BLOCK_TIME    portMAX_DELAY
+#if ( ( ipconfigUSE_RA != ipconfigDISABLE ) && ( ipconfigUSE_RA != ipconfigENABLE ) )
+    #error Invalid ipconfigUSE_RA configuration
 #endif
 
-/* When sending a UDP packet, a network buffer must be obtained.  This macro
- * will limit the maximum waiting time that is configured with e.g.
- * ipconfigSOCK_DEFAULT_SEND_BLOCK_TIME, or when setting the socket option
- * FREERTOS_SO_SNDTIMEO.
- * It serves as a protection against a possible dead-lock; a task waiting
- * for itself to release a network buffer.
- */
-#ifndef ipconfigUDP_MAX_SEND_BLOCK_TIME_TICKS
-    #define ipconfigUDP_MAX_SEND_BLOCK_TIME_TICKS    ( pdMS_TO_TICKS( 20U ) )
-#endif
+/*-----------------------------------------------------------------------*/
+
+/*#if ( ipconfigIS_ENABLED( ipconfigUSE_RA ) )*/
```

Comment:
Hey @holden-zenith 
Is there any reason we have the commented if at multiple places? MISRA 4.4 forbids to have comments to look like code

Reply:
oh yeah you can, they were there to remove configuration definitions that shouldn't be needed if something else is disabled but it turns out that is a little too messy to do at the moment with the current code. Maybe it can be looked at in a later PR.

Reply:
Sure. May be I will remove whatever MISRA4.4 is complaining for now.